code/drasil-printers/lib/Language/Drasil/Markdown/Helpers.hs

```diff
@@ -0,0 +1,127 @@
+paren t = text "(" <> t <> text ")" 
+
+-- | Angled brackets
+ang :: Doc -> Doc
+ang t = text "<" <> t <> text ">"
+
+-- | Bold text
+bold :: Doc -> Doc
+bold t = ast <> ast <> t <> ast <> ast
+
+-- | Italicized text
+em :: Doc -> Doc
+em t = ast <> t <> ast
+
+li, ul :: Doc -> Doc
+-- | List tag wrapper
+li = wrap "li"
+-- | Unordered list tag wrapper.
+ul = wrap "ul"
+
+-- | Helper for wrapping HTML lists
+wrap :: String -> Doc -> Doc
+wrap a = wrapGen hcat None a empty
+
+-- | Helper for setting up HTML tags
+-- ([Doc] -> Doc): Doc concatenation function.
+-- Variation: HTML tag attribute.
+-- String: HTML tag.
+-- Doc: HTML tag attribute value.
+-- Doc: Content being wrapped.
+wrapGen :: ([Doc] -> Doc) -> Variation -> String -> Doc -> Doc -> Doc
+wrapGen sepf None s _ = \x ->
+  let tb c = text $ "<" ++ c ++ ">"
+  in sepf [tb s, x, tb $ '/':s]
+wrapGen sepf Align s ti = \x ->
+  let tb c = text ("<" ++ c ++ " align=\"") <> ti <> text "\">"
+      te c = text $ "</" ++ c ++ ">"
+  in  sepf [tb s, x, te s]
+wrapGen sepf Id s ti = \x ->
+  let tb c = text ("<" ++ c ++ " id=\"") <> ti <> text "\">"
+      te c = text $ "</" ++ c ++ ">\n"
+  in  sepf [tb s, x, te s]
+
+-- | Helper for setting up section div
+divTag :: Doc -> Doc
+divTag l = wrapGen hcat Id "div" l empty
+
+-- | Helper for setting up div for defn heading
+defnHTag :: Doc -> Doc
+defnHTag = wrapGen vcat Align "div" (text "center")
+
+-- | Helper for setting up links to references
+reflink :: RefMap -> String -> Doc -> Doc
+reflink rm ref txt = sq txt <> paren rp
+  where
+    fn = maybe empty fp (lookup ref rm)
+    fp s = text $ "./" ++ s ++ ".md"
+    rp = fn <> text ("#" ++ ref)
+
+-- | Helper for setting up links to references with additional information.
+reflinkInfo :: RefMap -> String -> Doc -> Doc -> Doc
+reflinkInfo rm rf txt info = reflink rm rf txt <+> info
+
+-- | Helper for setting up links to external URIs
+reflinkURI :: Doc -> Doc -> Doc
+reflinkURI ref txt = if ref == txt then ang ref
+  else sq txt <> paren ref
+
+-- | Helper for setting up figures
+image :: Doc -> Doc -> Doc
+image f c =  text "!" <> reflinkURI f c $$ bold (caption c) <> text "\n"
+
+-- | Helper for setting up captions
+caption :: Doc -> Doc
+caption = wrapGen hcat Align "p" (text "center")
+
+-- | Helper for setting up headings with an id attribute.
+-- id attribute will only work for mdBook.
+heading ::  Doc -> Doc -> Doc
+heading t l = t <+> br (text "#" <> l)
+
+-- | Helper for setting up heading weights in mdBook.
+h :: Int -> Doc
+h n
+  | n < 1     = error "Illegal header (header weigth must be > 0)."
+  | n > 4     = error "Illegal header (header weight must be < 5)"
+  | n < 4     = h' 1
+  | otherwise = h' n
+
+-- | Helper for setting up heading weights in normal Markdown.
+h' :: Int -> Doc
+h' n
+  | n == 1    = text "#"
+  | n == 2    = text "##"
+  | n == 3    = text "###"
+  | n == 4    = text "####"
+  | otherwise = error "Illegal header (header weight must 0 < n < 5)"
```

Comment:
```suggestion
  | otherwise = error "Illegal header (header weight must be between 0 and 5)"
```
Just to be consistent with the other error messages which don't use `n`, although this could be approached in other ways

Reply:
Fixed in https://github.com/JacquesCarette/Drasil/pull/3823/commits/20ebf35520a935854188b3cab2b4600062620bc3.